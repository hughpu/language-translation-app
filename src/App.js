import axios from "axios";
import React, { Component, useState } from "react";
import "./App.css";


const Translator = () => {
  const [sourceText, setSourceText] = useState("你好！");
  const [targetText, setTargetText] = useState("hello!");
  const [error, setError] = useState(null);

  const handleSubmit = (event) => {
    event.preventDefault();

    const encodedParams = new URLSearchParams();
    encodedParams.append("q", sourceText);
    encodedParams.append("source", "zh");
    encodedParams.append("target", "en");
    
    const options = {
      method: 'POST',
      url: 'https://google-translate1.p.rapidapi.com/language/translate/v2',
      headers: {
        'content-type': 'application/x-www-form-urlencoded',
        'Accept-Encoding': 'application/gzip',
        'X-RapidAPI-Key': '02842eb1a6msh5ab52e77b1161d9p1d0c60jsnefebafd80afb',
        'X-RapidAPI-Host': 'google-translate1.p.rapidapi.com'
      },
      data: encodedParams
    };

    axios
      .request(options)
      .then(
        result => {
          setTargetText(result.data.data.translations[0].translatedText);
          setError(null);
        }
      ).catch(
        error => {
          setError(error);
        }
      )
  }

  const handleChange = (event) => {
    const value = event.target.value;
    setSourceText(value);
  }
  
  return (
    <form onSubmit={handleSubmit}>
      {error && <p color="r">{error.message}</p>}
      <div>
        <label htmlFor="sourceText">Source Chinese: </label>
        <input id="sourceText" type="text" onChange={handleChange} value={sourceText} />
      </div>
      <div>
        <label htmlFor="targetText">Target English: </label>
        <input id="targetText" type="text" value={targetText} readOnly/>
      </div>
      <button type="submit">Submit</button>
    </form>
  );
}

export default class App extends Component {
  render() {
    return <Translator />;
  }
}
