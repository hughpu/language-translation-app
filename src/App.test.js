import React from "react";
import { test, describe, expect } from "@jest/globals";
import renderer from "react-test-renderer";

import App from "./App";

describe('App', () => {
  test('Snapshot Match', () => {
    const app = renderer.create(<App />);
    expect(app.toJSON()).toMatchSnapshot();
  })
})
