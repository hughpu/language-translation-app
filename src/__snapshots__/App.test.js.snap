// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot Match 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <label
      htmlFor="sourceText"
    >
      Source Chinese: 
    </label>
    <input
      id="sourceText"
      onChange={[Function]}
      type="text"
      value="你好！"
    />
  </div>
  <div>
    <label
      htmlFor="targetText"
    >
      Target English: 
    </label>
    <input
      id="targetText"
      readOnly={true}
      type="text"
      value="hello!"
    />
  </div>
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
